File: Server/src/main/java/com/api/server/foodb/FoodTaxonomy.java
package com.api.server.foodb;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "food_taxonomies")
public class FoodTaxonomy {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "food_id")
    private Integer foodId;

    @Column(name = "ncbi_taxonomy_id")
    private Integer ncbiTaxonomyId;

    @Size(max = 255)
    @Column(name = "classification_name")
    private String classificationName;

    @Column(name = "classification_order")
    private Integer classificationOrder;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

}

File: DotNetServer/DotNetServer/Core/Entities/Foodb/FoodTaxonomy.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DotNetServer.Core.Entities.Foodb;

[Table("food_taxonomies")]
public class FoodTaxonomy
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("food_id")] public int? FoodId { get; set; }

    [Column("ncbi_taxonomy_id")] public int? NcbiTaxonomyId { get; set; }

    [Column("classification_name")]
    [StringLength(255)]
    public string? ClassificationName { get; set; }

    [Column("classification_order")] public int? ClassificationOrder { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }
}